OpenSim/Simulation/Model/Smith2016Ligament.h

```diff
@@ -0,0 +1,277 @@
+	// METHODS
+	//=============================================================================
+	public:
+		// Default Constructor
+		Smith2016Ligament();
+		Smith2016Ligament(PhysicalFrame& frame1, SimTK::Vec3 point1,
+			PhysicalFrame& frame2, SimTK::Vec3 point2);
+		Smith2016Ligament(PhysicalFrame& frame1, SimTK::Vec3 point1,
+			PhysicalFrame& frame2, SimTK::Vec3 point2,
+			double linear_stiffness, double reference_strain);
+
+		//-------------------------------------------------------------------------
+		//Outputs
+		//-------------------------------------------------------------------------
+		double getDynamicQuantities(const SimTK::State& state,
+			const std::string& channel) const {
+			if (channel == "force_spring") {
+				return getCacheVariableValue<double>(state, "force_spring");
+			}
+			else if (channel == "force_damping") {
+				return getCacheVariableValue<double>(state, "force_damping");
+			}
+			else if (channel == "force_total") {
+				return getCacheVariableValue<double>(state, "force_total");
+			}
+			else if (channel == "length") {
+				return getCacheVariableValue<double>(state, "length");
+			}
+			else if (channel == "lengthening_speed") {
+				return getCacheVariableValue<double>(state, "lengthening_speed");
+			}
+			else if (channel == "strain") {
+				return getCacheVariableValue<double>(state, "strain");
+			}
+			else if (channel == "strain_rate") {
+				return getCacheVariableValue<double>(state, "strain_rate");
+			}
+			else {
+				OPENSIM_THROW(Exception, "Output: " + channel + " is not a valid output for Smith2016Ligament.");
+			}
+		}
+
+
+		//--------------------------------------------------------------------------
+		// SET
+		//--------------------------------------------------------------------------
+		void setSlackLengthFromReferenceStrain(double referenceStrain, SimTK::State state);
+		void setSlackLengthFromReferenceForce(double referenceForce, SimTK::State state);
+
+		//--------------------------------------------------------------------------
+		// GET
+		//--------------------------------------------------------------------------
+		double getTension(const SimTK::State& s) const;
+
+		//--------------------------------------------------------------------------
+		// COMPUTATIONS
+		//--------------------------------------------------------------------------
+		virtual double computeMomentArm(const SimTK::State& s, Coordinate& aCoord) const ;
+		void computeForce(const SimTK::State& s,
+								  SimTK::Vector_<SimTK::SpatialVec>& bodyForces,
+								  SimTK::Vector& generalizedForces) const override;
+		double computePotentialEnergy(const SimTK::State& state) const override;
+				
+		
+		double computeSpringForce(double strain) const;
+		double computeSpringStrain(double force) const;
+		double computeReferenceForce(const SimTK::State& state) const;
+		double computeReferenceStrain(const SimTK::State& state) const;
+		double computeReferenceLength(SimTK::State state) const;
```

Comment:
`compute` is for a function that does a calculation. Use `get` if it's grabbing a cache value.